eth/api.go

```diff
@@ -217,18 +217,17 @@ func (s *PublicMinerAPI) SubmitWork(nonce rpc.HexNumber, solution, digest common
 // result[0], 32 bytes hex encoded current block header pow-hash
 // result[1], 32 bytes hex encoded seed hash used for DAG
 // result[2], 32 bytes hex encoded boundary condition ("target"), 2^256/difficulty
-func (s *PublicMinerAPI) GetWork() ([]string, error) {
+func (s *PublicMinerAPI) GetWork() (work [3]string, err error) {
 	if !s.e.IsMining() {
 		if err := s.e.StartMining(0, ""); err != nil {
-			return nil, err
+			return work, err
 		}
 	}
-	if work, err := s.agent.GetWork(); err == nil {
-		return work[:], nil
-	} else {
-		glog.Infof("%v\n", err)
+	if work, err = s.agent.GetWork(); err == nil {
+		return
 	}
-	return nil, fmt.Errorf("mining not ready")
+	glog.V(logger.Info).Infof("%v\n", err)
```

Comment:
Not sure this log level is correct. Should be `Debug` or `Warn` tbh depending on the severity... but info will imho just conflate it with general progress reports. PS: please remove the `\n`, it's not needed.